### DISCUSSION TITLE
Initial Release is there !!!

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/ha-anker-solix#11] [Announcements]
**Hello Anker friends and Solarbank owners**
You may have looked around since a while for a possible integration of your Solarbank or Inverter devices into Home Assistant, allowing you to monitor your energy flow and eventually integrate it into your energy management. 
The first integration is now available, based on the [anker-solix Python library](https://github.com/thomluther/anker-solix-api) I release few weeks ago. The integration supports translations which can easily be [expanded by the community](https://github.com/thomluther/hacs-anker-solix/discussions/12). It also comes with dynamic icons based on sensor states and optionally provides entity pictures for the supported Anker devices. 

Try the new integration and see if it fits your needs. Share your experience and observations with the community. 
The installation instruction can be found in the [README](https://github.com/thomluther/hacs-anker-solix/blob/main/README.md).
For detailed usage instructions, refer to the [INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md).

_**Keep in mind that you as a user bear the sole risks for the usage of the Integration, that may include loss of Manufacture's warranty or any damage that may have been caused by use of this integration or the underlying Api python library.**_ 

**Some heads up information before you ask:**
- If you do not find details you are interested in, look first in the sensors attributes. For instance the whole schedule json is provided with the actual device out power preset sensor
- No, there is no modification capability to set the home load (yet). This is complex since it can only be applied with a schedule, which is probably the whole but modified json schedule structure. It needs more testing and proper design for a user friendly utilization in HA
- No, there is no Solarbank temperature sensor unfortunately. I miss that too, but I have not seen this value in the known Api responses. I assume the temperature is provided only via Bluetooth like the real time data updates you can view in the solarbank details view of the Anker App.
- Yes, there are extra sensors that are not provided by the Anker cloud Api itself, like remaining battery energy and battery capacity attribute. They are provided/created by the Api python library to provide more useful information of your device.
- Be aware that the status descriptions which are provided also by the Api python library are based on experience. They may not always reflect the correct situation, in this case let me know with an example to correct this accordingly. Those sensors have the status code from the cloud Api as attribute. You should also know, that for example the charging_status code of 3 (= charging) is translated to three different status descriptions, depending on the other power metrics seen (charge, charge_priority and charge_bypass). The solarbank does not distinguish them when charging, however the descriptions should be more meaningful to users, reflecting a more accurate status condition.
- The Api library also converts some of the returned power fields since their meaning is not always identical and various weird values are presented in various situations, especially for the charging power. The Api library now converts/corrects the charging_power values to positive charging values and negative discharge values, giving a direct indication for the power direction. This can also easier be integrated into your HA power flow cards without the need of template sensors to indicate the power direction.
- The cloud Api may also sum up various values per reported device category, which are presented in the system device. For example if you have installed 2 Solarbanks, the System device will report the overall power values and SOC for both Solarbanks. Of course, those sensors are redundant if only 1 solarbank is installed, because they will show the same value as the individual device sensor. 
- If end device sensors are added by the Api library, they are not necessarily accumulated in the system again for the device category (like the battery energy). You can create a template or group sensor, if you want to sum or average individual device values for your dashboard.

**Comment by olivr2s** [comment]
Hello and thank you for this integration in HA which facilitates what I already had in place for my Anker batteries.
As much information as possible is already coming back and that's great.
Looking forward to being able to control the charging power in future updates, although managing schedules should not be that easy to manage.

**Reply by thomluther** [reply]
Thanks for the feedback.
Just FYI. The initial release is not ready yet, but HACS installation should work as documented.
But there will still be a couple of updates this week probably before I cut the first release.
I will announce in the HA forum when the initial release is ready

**Reply by olivr2s** [reply]
For my part, the installation went very well, the configuration was very simple and the information was provided immediately.
For the moment, it is working well and the real-time information seems correct.
I will see if I can translate it into French.

**Comment by manisflight** [comment]
Installation is okey, but after when i have to login in my account, i receive: Unknown error occurred, i cant login to Cloud

**Reply by manisflight** [reply]
I try again today, now it rocks! 
Thank you for your good Work.

**Comment by Pouda41** [comment]
Thank you for this good job, and for the very good documentation for newbies !

**Comment by FloFR92** [comment]
Hi All,
Thansk for this addon it's really nice to have ths solarbank and MI90 overview in HA.
Do you know if we can change the setting of the energy dispatch program set in the Anker App ?
I would like to adapt the plan with the sun production and my personal consumtion.

thanks for your feedbacks

**Comment by clemone210** [comment]
> Hi All,
> 
> Thansk for this addon it's really nice to have ths solarbank and MI90 overview in HA.
> 
> Do you know if we can change the setting of the energy dispatch program set in the Anker App ?
> 
> I would like to adapt the plan with the sun production and my personal consumtion.
> 
> 
> 
> thanks for your feedbacks

As in the project is mentioned, this is not possible (yet). 
Maybe it can be integrated in the future.

**Reply by thomluther** [reply]
This is in progress for the next release:
https://github.com/thomluther/hacs-anker-solix/milestone/2

**Comment by spaxxilein** [comment]
After adding the integration VIA HACS and restarting i try to add the integration. After inserting Username (EMAIL) and Password, select country DE and accept the TOS and hit save i only get "Unknown error occurred". Any ideas?

EDIT: i am using a second account which has shared access to the SOLIX E1600 from the main account i use for APP.

**Reply by thomluther** [reply]
Yeah, have you checked in the app that you accepted the invite? You need to follow the described process completely.
When you can use the second account in the Anker App, the integration should also be able to use it.
Typically there should be an authentication error reflected however.
Maybe that is one of the corner cases I did not see.
You can also enabled debugging for the anker_solix integration an provide the integration logs. Not sure if that will show more why you cannot login...

**Comment by thealkly** [comment]
really great work @thomluther !  Thank you very much! After installing the PV & Mounting with Solar Banks yesterday for a new yt video, the setup in HA was really super fast. I was expecting a much more difficult setup :D 
It helped me to understand what exactly the solar bank is doing (charging, bypass, etc).

@spaxxilein I guess it worked with the second account in the app? i followed the recommendation with a small deviation. I first registered my second account via the anker website. Found out there is no way to accept the invite into the BKA System. Later on i found out that the anker app user mngt is somehow different. i had to reset my password. But after reset & login into the app, my invite worked. not sure if this is relevant.

**Reply by thomluther** [reply]
Hi @thealkly , I think I know your YT channel.
Generally speaking I'm still bit concerned how public we should make the integration since Anker Support already expressed that they have no plans to provide any smart home integration for Anker Solix devices. Since the Api library used by the integration is not official, I have no clue what standing Anker has on unofficial use of their cloud Api....
In worst case, they could modiify login authentication / encryption mechanism in a future release and break any non App client for the cloud Api...

**Comment by Zordandi** [comment]
Many thanks to you @thomluther for this great work!
I was able to install everything very easily and everything runs like clockwork.

**Comment by Seiher** [comment]
Hi, wen I will aktivste the Integration, I become thiy Massage: 
Fehler
Der Konfigurationsfluss konnte nicht geladen werden: 500 Internal Server Error Server got itself in trouble

**Reply by thomluther** [reply]
You should open an issue and provide debug log data.
With only such a statement, nobody can help you.

**Reply by Seiher** [reply]
thank you for your Answer, i fix the problem
